### PR TITLE
allow overriding range if called on visual selection

### PR DESCRIPTION
**Pull request by Konfekt**
Check if the start and end line of range equals that of visual selection.
If so, take the visual selection;
otherwise the supplied range

### CODE REVIEW

**Comment by madox2** [comment]
How is this improvement useful in the usage?

**Comment by Konfekt** [comment]
Sometimes text is selected, when the visual selection markers are inserted.
If one changes the range, then currently this explicitly given range is not respected. (To my surprise, the visual selection was still used)

**Comment by Konfekt** [comment]
This is just a safety check that was missing. It crossed my mind at some point, but was forgotten. Till I stumbled upon it.

**Comment by Konfekt** [comment]
So, for example, this ensures that when the user enters`:%AIEdit`, then the whole buffer should be processed, not the visual selection.

File: autoload/vim_ai.vim [hunk 1]
```diff
@@ -1,4 +1,13 @@
 call vim_ai_config#load()
+if &compatible || exists('g:loaded_vim_ai')
+    finish
+endif
+
+function! s:vim_ai()
+    
+endfunction
+
+let g:loaded_vim_ai = 1
```
Comment (madox2) [inline autoload/vim_ai.vim:10]:
```suggestion
```

Comment (Konfekt) [reply autoload/vim_ai.vim]: Thank you, the last quick fix unintentionally introduced boilerplate code; now removed.

**Comment by madox2** [comment]
Thank you! Looks good

**Pull request merged**